File: eb-application/src/main/java/com/ebnbin/ebapplication/context/EBApplication.java
package com.ebnbin.ebapplication.context;

import android.app.Application;

import com.ebnbin.ebapplication.net.NetHelper;

/**
 * Base {@link Application}.
 */
public abstract class EBApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        initNetHelper();
    }

    private void initNetHelper() {
        NetHelper.init();
    }
}

File: eb-application/src/main/java/com/ebnbin/ebapplication/context/EBApplication.kt
package com.ebnbin.ebapplication.context

import android.app.Application
import com.ebnbin.eb.util.EBUtil
import com.ebnbin.ebapplication.net.NetHelper

/**
 * Base [Application].
 */
abstract class EBApplication : Application() {
    override fun onCreate() {
        super.onCreate()

        initEBUtil()
        initNetHelper()
    }

    private fun initEBUtil() {
        EBUtil.init(applicationId())
    }

    /**
     * Returns application's `BuildConfig.APPLICATION_ID`.
     */
    protected abstract fun applicationId(): String

    private fun initNetHelper() {
        NetHelper.init()
    }
}
